### ISSUE TITLE
[QUESTION] How to delete widget from widget list?

### ISSUE THREAD

**Issue by Der-Flip** [fibelatti/photo-widget#88]
Hi, this is a bit of a weird question...

I have been testing out the app and so far absolutely loving it!

I have added several widgets from the Homescreen and later deleted them from the home screen as well. Now they are [still being listed in the _My Widgets_ list](https://github.com/user-attachments/assets/ee1ee1f5-51eb-482d-b13d-2460082f6929).

I've tried long tap to delete but I don't have an option there and when selecting edit there isn't an option either which would let me remove it from the list. Am I missing something or is that not a function yet?

**Comment by fibelatti** [comment]
Hi @Der-Flip, thanks for reaching out and I'm glad to hear you're liking the app so far!

Have you used different launchers and placed widgets on them by any chance? That list displays widgets placed on all launchers, not only your active one.

There's no option to delete a widget from within the app because that action can only be taken by the user from the home screen, so apps cannot do it. You have two options to work around the issue you're seeing:

1. Try editing them, adding a photo and then placing them on the home screen. Next you should be able to remove them if you so wish;
2. Alternatively, you can long press the app icon and tap to view app info. Look for the storage option and clear the app storage. **Beware that this will delete all current widgets and preferences** (like widget defaults), so it can be a bit inconvenient 

Let me know if either of these work.

**Comment by Der-Flip** [comment]
Hi, thanks for responding 

So I tried adding a new source / photo to the one displayed in red, then confirming at the bottom by placing on home screen, but they never appeared on the home screen. For testing purposes I then created another new widget, confirmed it showed up in the _my widgets_ list, then deleted and it was removed from the list as well. So something about the earlier ones seems to be bugged. I remember creating the widgets before having opened the app itself, could that be connected?

I then tried to delete all data (Settings, Apps, manage apps, Material Photo Widget, clear data, clear all data). Contrary to expectations this did not remove the widget that was (is) currently on the Homescreen but in the app they are all shown with the red 'missing' icon.

I only ever used the phone as stock as it was (android 10, miui12). (First android after blackberry meant to be a temporary phone, wanted to experiment once I had a new primary daily driver).

I'll try reinstalling next. :'D

Thanks for your help though!

**Comment by fibelatti** [comment]
@Der-Flip thanks for sharing the details. The app has some issues with MIUI but I haven't gotten to the bottom of it as I don't have a device with it myself, making it tricky to troubleshoot.

>  I remember creating the widgets before having opened the app itself, could that be connected?

By this you mean creating the widgets directly from the home screen, correct? That's interesting to know. So far I thought that the issues were tied to creating a widget from the app and requesting to add them to the home screen.

> I then tried to delete all data (Settings, Apps, manage apps, Material Photo Widget, clear data, clear all data). Contrary to expectations this did not remove the widget that was (is) currently on the Homescreen but in the app they are all shown with the red 'missing' icon.

That's interesting as well and appears to be particular to MIUI as other ROMs do not behave like that. When clearing the app data all placed widgets would be removed. I'll try researching this a little bit to see if there's anything I can do on my end.

Good to hear that reinstalling works, at least that did the trick 😅

**Comment by Der-Flip** [comment]
> By this you mean creating the widgets directly from the home screen, correct?

Correct basically
1. Install app
2. Tap and hold home screen
3. Select "add widget"
4. Add widget.

All of this without opening the app after install and before adding a widget.

I haven't done much more experimenting yet tbh. If you need any more details I'm happy to see if I can help, just note, I'm not super savvy, so if you need me to check something, please take it slow and eli5 :'D

Tbh, I don't really mind the issues too much, as I will only have one widget anyways so if I need to reinstall every once in a while (once in several years) to clear the my widgets list, I can perfectly live with that :'D
